### PR TITLE
Add directive support for variable injection

### PR DESCRIPTION
**Pull request by ckoberlein**
- Add values attribute that accepts JSON data for variable injection
 - Conditionally interpolate translated string if it contains double
   curly braces indicating Angular expression
 - Update README with documentation on usage for new feature

### CODE REVIEW

**Comment by srl295** [comment]
Fixes #16

File: README.md [hunk 1]
```diff
@@ -140,7 +140,23 @@ Example:
...
 
+Example translation string:
+
+`"DATA-KEY-WITH-VARIABLE" : "Welcome {{user}}"`
+
+Example usage:
+
+```html
+
+    <gp-translate key="DATA-KEY-WITH-VARIABLE" bundle="someBundle" values="{{controllerAsVariable.userName}}"/>
```
Comment (srl295) [inline README.md]: does this assume `controllerAsVariable.userName` is a map? Or does it just assume the only item will be substituted?

File: src/gp-angular.js [hunk 2]
```diff
@@ -559,21 +559,22 @@ module.provider('GlobalizationPipelineService', [ function() {
  *  <span gp-translate="aKey"></span>
  *  <span gp-translate="aKey" lang="es"></span>
  */
-module.directive('gpTranslate', ['$log', 'GlobalizationPipelineService', function($log, gp) {
+module.directive('gpTranslate', ['$log', '$interpolate', 'GlobalizationPipelineService', function($log, $interpolate, gp) {
```
Comment (srl295) [inline src/gp-angular.js]: Does `$interpolate` need to be "enabled" anywhere as a module? And, are there docs we can point to with the full syntax?

File: src/gp-angular.js [hunk 3]
```diff
@@ -597,6 +598,23 @@ module.directive('gpTranslate', ['$log', 'GlobalizationPipelineService', functio
                     if(DEBUG) {
                         $log.debug("[gp directive] promise resolved...value["+string+"]");
                     }
+
+                    // Check if the translation string has a variable expression
+                    var interpolateFn = $interpolate(string, true);
```
Comment (srl295) [inline src/gp-angular.js]: Can this fail? I wonder if we should check `scope.values` first (that way one could use `values: false` to disable interpolation).  Again, not sure of the full syntax.

File: src/gp-angular.js [hunk 4]
```diff
@@ -597,6 +598,23 @@ module.directive('gpTranslate', ['$log', 'GlobalizationPipelineService', functio
                     if(DEBUG) {
                         $log.debug("[gp directive] promise resolved...value["+string+"]");
                     }
+
+                    // Check if the translation string has a variable expression
+                    var interpolateFn = $interpolate(string, true);
+                    if (scope.values && interpolateFn) {
+                        try {
+                            // Use the JSON data to populate the variables
+                            var dataContext = JSON.parse(scope.values);
```
Comment (srl295) [inline src/gp-angular.js]: could `values` already be a map here? Should this be parsed only if `typeof(scope.values) === 'string'`?

File: src/gp-angular.js [hunk 5]
```diff
@@ -597,6 +598,23 @@ module.directive('gpTranslate', ['$log', 'GlobalizationPipelineService', functio
...
+                        try {
+                            // Use the JSON data to populate the variables
+                            var dataContext = JSON.parse(scope.values);
+                            var interpolatedString = interpolateFn(dataContext);
+                            if (DEBUG) {
+                                $log.debug("interpolatedString: " + interpolatedString);
+                            }
+                            string = interpolatedString;
+                        } catch (e) {
+                            $log.error("Error evaluating optional values: " + e);
```
Comment (srl295) [inline src/gp-angular.js]: --> `error interpolating…` ?

File: src/gp-angular.js (hunk 2 above)
Comment (ckoberlein) [inline src/gp-angular.js]: `$interpolate` is a base Angular service, so no enabling needed.  It just needs to be injected into wherever you're planning to use it as seen here.

Would you like the doc link as a comment above this?
https://docs.angularjs.org/api/ng/service/$interpolate

File: README.md (hunk 1 above)
Comment (ckoberlein) [inline README.md]: `{{controllerAsVariable.userName}}` would get substituted/interpolated prior to being processed by the directive.  The only requirement is that whatever is in there is valid JSON.

In my testing, the controller had a variable declared like this:

`userName = { user : "John Doe" };`

Any suggestion on improving the documentation to make it more clear?

Comment (srl295) [inline README.md]: Probably including the controller sample would be good.  Also, some tests ( see #18 ) could be good.

File: src/gp-angular.js (hunk 2 above)
Comment (srl295) [inline src/gp-angular.js]: not as a comment, but in the readme.

Comment (ckoberlein) [inline src/gp-angular.js]: Done

File: src/gp-angular.js (hunk 3 above)
Comment (ckoberlein) [inline src/gp-angular.js]: No, the second parameter ensures that only strings that contain `{{stuff}}` will set a value to `interpolateFn`.  If there are no expressions to be interpolated, `interpolateFn` will be `undefined`.

So my `if` check right below will only continue if the user has passed values and the string would accept them.  If one or the other is not true, then the translated string is returned as-is.

File: src/gp-angular.js (hunk 5 above)
Comment (ckoberlein) [inline src/gp-angular.js]: Done

**Comment by srl295** [comment]
tests LGTM

**Pull request merged**